design/fast_finality.md

```diff
@@ -0,0 +1,273 @@
+  * The sequencer is not able to "rewrite history" (or is strongly disincentivised from doing so), even for soft-final
+    transactions
+  * Soft finality does not break the mechanism used by some smart contracts (e.g. SushiSwap) of using `block.number` as 
+    a rudimentary clock
+* Censorship resistance
+  * End-users have a mechanism to force the sequencer to include their transactions (possibly at higher cost and slower 
+    finality)
+  * The sequencer distributes all soft-finalised transactions to all nodes
+  * Nodes can easily prove whether a given soft-finalised transaction was hard-finalised, and in the correct order
+* Value-extraction resistance
+  * The sequencer cannot precompute the effects of running a given set of transactions without committing to that set 
+    of transactions (e.g. by running a single transaction, then `eth_call`ing to see how it has affected a smart 
+    contract's state)
+* Cost
+  * L1 transaction costs can be driven lower at the expense of extending the hard-finality window
+  * The network can be configured to only publish rollups to the L1 every `x` blocks, where `x` >> 1
+  * The network can be configured to produce light batches (see below) at a higher frequency than L1 blocks
+* User/dev experience
+  * The responses to RPC calls reflect the soft-finalised transactions, and not just the hard-finalised transactions
+* Resilience
+  * Failover does not require a governance action
+  * During failover, it is acceptable to break the one-second soft-finality guarantee 
+  * During failover, it is acceptable to drop transactions that have not been soft-finalised
+
+## Assumptions
+
+* There is a single aggregator that is also the genesis node. This node is known as the *sequencer*
+* The enclave has a start-up delay that exceeds the production time for light batches (see below)
+
+## Design
+
+### Sequencer identity
+
+The sequencer's identity is given in the management contract on the L1 as a set of enclave attestations. This allows 
+other nodes to verify that the light batches and rollups are created by the sequencer. We use a set instead of a 
+single attestation to allow faster failover in the case of a sequencer node crashing (see the section `Resilience`, 
+below).
+
+Each attestation matches one of the sequencer's enclaves, and contains the hash of that enclave's key. Since the
+attestations are unique per machine, the enclaves cannot be impersonated. The foundation admin will then whitelist
+these attestations. If one of the sequencer's enclaves goes down irrecoverably, the old attestation can no longer be 
+used (since it is tied to the machine). A replacement attestation can then be added and whitelisted to induct a 
+replacement enclave into the sequencer's cluster, following the same process.
+
+### Production of light batches
+
+A light batch is produced on the required cadence to meet the network's soft-finality window of one second. Only the 
+sequencer produces light batches.
```

Comment:
would be good to note that this will be enforced in the enclave code itself. Basically, unless you're whitelisted, you can't produce LBs